### ISSUE TITLE
Submodule context from Unstaged changes section

### ISSUE THREAD

**Issue by andyturk** [magit/magit#3385]
It would be useful if commands in the submodule popup could use context from a selected submodule in the **Unstaged changes** section of a magit status buffer. This would be similar to what happens when a submodule is selected in the **Modules overview** section.

E.g., suppose I have the following in my status buffer:
```
Unstaged changes (1)
modified   submodule/foobar

Modules overview (1)
submodule/foobar master                                   aabbccdd
```
Currently, selecting the `submodule/foobar` line in **Modules overview** will default to using that info for submodule commands. However, selecting the `submodule/foobar` line in **Unstaged changes** prompts for more information.

This improvement would help alleviate confusion when the **Modules overview** section is not visible (which is the default).

**Comment by tarsius** [comment]
True, but due to implementation details this is currently not possible without adding a major kludge. The problem here is that these sections have type `file`, not type `module`.

I am just about to move from a phase when I tried to implement as many things as possible even if that required some hacks to a phase focuses on improving abstractions so that hacks become less necessary.

If you had opened this issue about two weeks ago, then I would probably have addressed this by now, but I am afraid now you will have to wait a few months.

**Comment by tarsius** [comment]
Implemented in #3590.
